plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -898,13 +946,18 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 			self->fw_key_addr[1] = 0xE000;
 			self->code_size[0] = self->code_size[1] = self->fw_data_total_count;
 		}
+		self->dynamic_ts.bonding =
+		    _fu_genesys_revert_hex_char_to_int(self->dynamic_ts.bonding);
```

Comment:
can you create a local variable bonding and convert it to int with `bonding = g_ascii_xdigit_value(self->dynamic_ts.bonding));`

Reply:
Done.

Reply:
ditto.